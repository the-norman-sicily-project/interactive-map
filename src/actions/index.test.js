import configureStore from 'redux-mock-store';

import * as actions from './index';

const mockStore = configureStore();
const store = mockStore();

describe('actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe('initMap', () => {
    test('Dispatches init map action', () => {
      store.dispatch(actions.initMap());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('loadSitesBegin', () => {
    test('Dispatches load sites begin action', () => {
      store.dispatch(actions.loadSitesBegin());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('loadSitesSuccess', () => {
    test('Dispatches load sites success action', () => {
      store.dispatch(actions.loadSitesSuccess({ features: [1, 2, 3] }));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('loadSitesFAILURE', () => {
    test('Dispatches load sites failure action', () => {
      store.dispatch(actions.loadSitesFailure(new Error('error occurred')));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
