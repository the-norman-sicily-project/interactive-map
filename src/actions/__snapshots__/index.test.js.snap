// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions initMap Dispatches init map action 1`] = `
Array [
  Object {
    "type": "INIT_MAP",
  },
]
`;

exports[`actions loadSitesBegin Dispatches load sites begin action 1`] = `
Array [
  Object {
    "type": "LOAD_SITES_BEGIN",
  },
]
`;

exports[`actions loadSitesFAILURE Dispatches load sites failure action 1`] = `
Array [
  Object {
    "error": [Error: error occurred],
    "type": "LOAD_SITES_FAILURE",
  },
]
`;

exports[`actions loadSitesSuccess Dispatches load sites success action 1`] = `
Array [
  Object {
    "sites": Array [
      Number {
        "id": "site_1",
      },
      Number {
        "id": "site_2",
      },
      Number {
        "id": "site_3",
      },
    ],
    "type": "LOAD_SITES_SUCCESS",
  },
]
`;
